webapp/src/cloud.ts

```diff
@@ -382,28 +393,27 @@ async function syncAsyncInternal(hdrs?: Header[]): Promise<Header[]> {
                         // No local changes, download latest.
                         const newHdr = await fromCloud(local, remoteFile);
                         pxt.tickEvent(`identity.sync.noConflict.localProjectUpdatedFromCloud`)
+                    } else if (local.cloudVersion === remoteFile.version) {
+                        // Local has unpushed changes, push them now
+                        pxt.debug(`local project '${local.name}' has changes that will be pushed to the cloud.`)
+                        const newHdr = await toCloud(local, remoteFile.version);
+                        U.assert(!!newHdr, `Failed to save ${local.id} to the cloud.`);
+                        pxt.tickEvent(`identity.sync.noConflict.localProjectUpdatedToCloud`)
                     } else {
-                        // Possible conflict.
-                        const conflictStr = `conflict found for ${projShorthand}. Last cloud change was ${agoStr(remoteFile.header.modificationTime)} and last local change was ${agoStr(local.modificationTime)}.`
+                        // Conflict. Local changes exist and we aren't on the latest cloud version.
                         if (local.modificationTime > remoteFile.header.modificationTime) {
```

Comment:
>modificationTime [](start = 34, length = 16)

Is this a client-generated timestamp? If so, it can't be relied upon.

Reply:
it is client generated but in this case all it's informing is which tick to send since we always do the same copy solution.